packages/ember-template-compiler/lib/plugins/transform-resolutions.ts

```diff
@@ -149,7 +149,9 @@ export default function transformResolutions(env: EmberASTPluginEnvironment): AS
 }
 
 function isLocalVariable(node: AST.PathExpression, hasLocal: (k: string) => boolean): boolean {
-  return !node.this && node.parts.length === 1 && hasLocal(node.parts[0]!);
+  return (
+    !(node.head.type === 'ThisHead') && node.tail.length === 1 && hasLocal(node.head.original!)
```

Comment:
Why the ! on `.original`? It should always exist?

Reply:
yes, but nothing complained about it. It's been removed now.